AutomatedTesting/Gem/PythonTests/AWS/common/constants.py

```diff
@@ -12,7 +12,7 @@
 ASSUME_ROLE_ARN = os.environ.get('ASSUME_ROLE_ARN', 'arn:aws:iam::645075835648:role/o3de-automation-tests')
 # Name of the AWS project deployed by the CDK applications
 AWS_PROJECT_NAME = os.environ.get('O3DE_AWS_PROJECT_NAME').upper() if os.environ.get('O3DE_AWS_PROJECT_NAME') else \
-    (os.environ.get('BRANCH_NAME', '') + '-' + os.environ.get('PIPELINE_NAME', '') + '-' + platform.system()).upper()
+    (os.environ.get('BRANCH_NAME', '') + '-' + os.environ.get('PIPELINE_NAME', '')[:25] + '-' + platform.system()).upper()

```

Comment:
nit: 25 is magic number, consider at least referencing in comments why 25

Reply:
Would there be a issue here if the branch name also exceeds the character limit?